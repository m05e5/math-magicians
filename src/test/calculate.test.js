// import operate from '../logic/operate';
// import { Link } from 'react-router-dom';
// import renderer from 'react-test-renderer';
import calculate from '../logic/calculator';

describe('checking operators', () => {
  let myObject = {
    total: null,
    next: null,
    operation: null,
  };

  it('test the result of + operation', () => {
    myObject = {
      total: 5,
      next: 9,
      operation: '+',
    };

    const result = calculate(myObject, '=');

    expect(result).toMatchSnapshot({
      total: '14',
      next: null,
      operation: null,
    });
  });
  it('test the result of - operation', () => {
    myObject = {
      total: 5,
      next: 9,
      operation: '-',
    };

    const result = calculate(myObject, '=');

    expect(result).toMatchSnapshot({
      total: '-4',
      next: null,
      operation: null,
    });
  });
  it('test the result of +/-', () => {
    myObject = {
      total: 5,
      next: 9,
      operation: '-',
    };

    const result = calculate(myObject, '+/-');

    expect(result).toMatchSnapshot({
      next: '-9',
    });
  });
});

// it('renders correctly', () => {
//   const tree = renderer
//     .create(<Link to="/calculator" className="navLink">calculator</Link>)
//     .toJSON();
//   expect(tree).toMatchSnapshot();
// });
