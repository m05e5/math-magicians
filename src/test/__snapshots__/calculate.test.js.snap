// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checking operators test the result of + operation 1`] = `
Object {
  "next": null,
  "operation": null,
  "total": "14",
}
`;

exports[`checking operators test the result of +/- 1`] = `
Object {
  "next": "-9",
}
`;

exports[`checking operators test the result of - operation 1`] = `
Object {
  "next": null,
  "operation": null,
  "total": "-4",
}
`;
